src/main/java/org/motechproject/newebodac/mapper/ExtraFieldMapper.java

```diff
@@ -1,34 +1,124 @@
 
+  @Mapping(target = "visitTypeId", source = "visitType.id")
+  default void update(ExtraFieldDto extraFieldDto, @MappingTarget ExtraField extraField) {
+    FieldType fieldType = FieldType.valueOf(extraFieldDto.getFieldType());
+    setValue(extraField, extraFieldDto);
+    extraField.setFieldType(fieldType);
+    extraField.setName(extraFieldDto.getName());
+  }
+
   @Override
-  @Mapping(target = "id", ignore = true)
-  @Mapping(target = "vaccinee", source = "vaccineeId")
-  ExtraField fromDto(ExtraFieldDto extraFieldDto);
+  default ExtraField fromDto(ExtraFieldDto extraFieldDto) {
+    ExtraField extraField = new ExtraField();
+    FieldType fieldType = FieldType.valueOf(extraFieldDto.getFieldType());
+    if (fieldType != null) {
+      setValue(extraField, extraFieldDto);
+      extraField.setFieldType(fieldType);
+      extraField.setName(extraFieldDto.getName());
+    }
+    return extraField;
+  }
 
   @Override
-  @Mapping(target = "vaccineeId", source = "vaccinee.id")
-  @Mapping(target = "visitId", source = "visit.id")
-  @Mapping(target = "siteId", source = "site.id")
-  @Mapping(target = "groupId", source = "group.id")
-  @Mapping(target = "personId", source = "person.id")
+  @Mapping(target = "value", expression = "java(toValue(extraField))")
   ExtraFieldDto toDto(ExtraField extraField);
 
+  /**
+   * Gets fields related to fieldType and casts it to string.
+   * @param extraField Mapped entity.
+   * @return String value of related to fieldType field.
+   */
+  default String toValue(ExtraField extraField) {
+    String result;
+    switch (extraField.getFieldType()) {
+      case TEXT:
+        result = extraField.getTextVal();
+        break;
+      case LONG_TEXT:
+        result = extraField.getLongTextVal();
+        break;
+      case INTEGER:
+        result = Objects.toString(extraField.getIntVal(), null);
+        break;
+      case FLOAT:
+        result = Objects.toString(extraField.getFloatVal(), null);
+        break;
+      case BOOLEAN:
+        result = Objects.toString(extraField.getBoolVal(), null);
+        break;
+      case DATE:
+        result = Objects.toString(extraField.getDateVal(), null);
+        break;
+      case DATE_TIME:
+        result = Objects.toString(extraField.getDatetimeVal(), null);
+        break;
+      default:
+        return null;
+    }
+    return result;
+  }
+
+  @Override
+  List<ExtraFieldDto> toDtos(Iterable<ExtraField> extraFields);
+
+  Set<ExtraField> fromDtos(Iterable<ExtraFieldDto> extraFieldDtos);
+
+  private void setValue(ExtraField extraField, ExtraFieldDto extraFieldDto) {
+    FieldType fieldType = FieldType.valueOf(extraFieldDto.getFieldType());
+    switch (fieldType) {
+      case TEXT:
+        extraField.setTextVal(extraFieldDto.getValue());
+        break;
+      case LONG_TEXT:
+        extraField.setLongTextVal(extraFieldDto.getValue());
+        break;
+      case INTEGER:
+        extraField.setIntVal(Integer.parseInt(extraFieldDto.getValue()));
```

Comment:
the parse methods are not null safe, you should check if the value is not blank first